ERCS/erc-7920.md

```diff
@@ -0,0 +1,358 @@
+    ['DATA', 'DATA'], // First message proof (array of 32 byte hex strings)
+    ['DATA', 'DATA'], // Second message proof
+    ...
+  ]
+}
+```
+
+##### Example
+
+Request:
+
+```shell
+curl -X POST --data '{"jsonrpc":"2.0","method":"eth_signTypedData_v5","params":["0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826", [{"types":{"EIP712Domain":[{"name":"name","type":"string"},{"name":"version","type":"string"},{"name":"chainId","type":"uint256"},{"name":"verifyingContract","type":"address"}],"Person":[{"name":"name","type":"string"},{"name":"wallet","type":"address"}],"Mail":[{"name":"from","type":"Person"},{"name":"to","type":"Person"},{"name":"contents","type":"string"}]},"primaryType":"Mail","domain":{"name":"Ether Mail","version":"1","chainId":1,"verifyingContract":"0xCcCCccccCCCCcCCCCCCcCcCccCcCCCcCcccccccC"},"message":{"from":{"name":"Cow","wallet":"0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826"},"to":{"name":"Bob","wallet":"0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB"},"contents":"Hello, Bob!"}}, {"types":{"EIP712Domain":[{"name":"name","type":"string"},{"name":"version","type":"string"},{"name":"chainId","type":"uint256"},{"name":"verifyingContract","type":"address"}],"Transfer":[{"name":"amount","type":"uint256"},{"name":"recipient","type":"address"}]},"primaryType":"Transfer","domain":{"name":"Ether Mail","version":"1","chainId":1,"verifyingContract":"0xCcCCccccCCCCcCCCCCCcCcCccCcCCCcCcccccccC"},"message":{"amount":"1000000000000000000","recipient":"0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB"}}]],"id":1}'
+```
+
+Result:
+
+```JavaScript
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": {
+    "signature": "0x4355c47d63924e8a72e509b65029052eb6c299d53a04e167c5775fd466751c9d07299936d304c153f6443dfa05f40ff007d72911b6f72307f996231605b915621c",
+    "merkleRoot": "0x7de103665e21d6c9d9f82ae59675443bd895ed42b571c7f952c2fdc1a5b6e8d2",
+    "proofs": [
+      ["0x4bdbac3830d492ac3f4b0ef674786940fb33481b32392e88edafd45d507429f2"],
+      ["0x95be87f8abefcddc8116061a06b18906f32298a4644882d06baff852164858c6"]
+    ]
+  }
+}
+```
+
+## Rationale
+
+### UX Improvement
+
+A single signature that covers multiple messages
+
+### Isolated Verification
+
+Independent verification of messages without knowledge of others
+
+### Human-readable
+
+This ERC preserves the readability benefits of EIP-712. Giving wallets and users insight into what is being signed.
+
+### Efficient verification on-chain
+
+`_verifyMerkleProof` has a runtime of `O(log2(N))` where N is the number of messages that were signed.
+
+When N=1, merkleRoot **must** equal keccak256(encode(m₁)) and proofs MUST be empty arrays
+
+### Improved Wallet Security
+
+Certain messages signed in isolation may appear harmless but combined with may be harmful. Giving the full list of messages to users could help them better navigate their experience.
+
+### Flexible Verification Modes
+
+Applications can require combination of messages be signed together to enhance security.
```

Comment:
All of these points seem to fit more in the Motivation section rather than the Rationale. Motivation should justify the proposal as a whole, while the Rationale should explain decisions you made within the document. My favourite analogy is:

> Motivation: We need to build a shed because...
> Rationale: We chose to paint the shed red because...